apps/game_backend/lib/game_backend/curse_of_mirra/quests.ex

```diff
@@ -109,4 +111,56 @@ defmodule GameBackend.CurseOfMirra.Quests do
       Repo.transaction(multi)
     end
   end
+
+  def get_google_user_daily_quests_completed(%GoogleUser{
+        arena_match_results: arena_match_results,
+        user: %User{daily_quests: daily_quests}
+      }) do
+    Enum.reduce(daily_quests, [], fn daily_quest, acc ->
+      if completed_daily_quest?(daily_quest, arena_match_results) do
+        [daily_quest | acc]
+      else
+        acc
+      end
+    end)
+  end
+
+  #####################
+  #      helpers      #
+  #####################
+
+  defp parse_comparator("equal"), do: &Kernel.==/2
+  defp parse_comparator("distinct"), do: &Kernel.!=/2
+  defp parse_comparator("greater"), do: &Kernel.>/2
+  defp parse_comparator("greater_or_equal"), do: &Kernel.>=/2
+  defp parse_comparator("lesser"), do: &Kernel.</2
+  defp parse_comparator("lesser_or_equal"), do: &Kernel.<=/2
+  defp parse_comparator(comparator), do: raise("Comparator not implemented yet #{comparator}")
+
+  defp accumulate_objective_progress_by_scope("day", value), do: value
+  defp accumulate_objective_progress_by_scope("match", _value), do: 1
+
+  defp completed_daily_quest?(%DailyQuest{quest: %Quest{} = quest}, arena_match_results) do
+    progress =
+      Enum.filter(arena_match_results, fn arena_match_result ->
+        Enum.all?(quest.conditions, fn condition ->
+          type = String.to_atom(condition["match_tracking_field"])
```

Comment:
Can we use `String.to_existing_atom/1` here?

Reply:
Sure thing! I'll make the change but i'm not sure if we're saving something doing this, `String.to_atom` doesn't create new atoms if they already exist, and since this isn't user input and come from the config file we shouldn't have an atom overflow